commit 71afa85e798ebc81054164c947b189d163c92b30
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Sat Mar 16 15:54:16 2013 -0400

    SUNRPC: Missing module alias for auth_rpcgss.ko
    
    Commit f344f6df "SUNRPC: Auto-load RPC authentication kernel
    modules", Mon Mar 20 13:44:08 2006, adds a request_module() call
    in rpcauth_create() to auto-load RPC security modules when a ULP
    tries to create a credential of that flavor.
    
    In rpcauth_create(), the name of the module to load is built like
    this:
    
            request_module("rpc-auth-%u", flavor);
    
    This means that for, say, RPC_AUTH_GSS, request_module() is looking
    for a module or alias called "rpc-auth-6".
    
    The GSS module is named "auth_rpcgss", and commit f344f6df does not
    add any new module aliases.  There is also no such alias provided in
    /etc/modprobe.d on my system (Fedora 16).  Without this alias, the
    GSS module is not loaded on demand.
    
    This is used by rpcauth_create().  The pseudoflavor_to_flavor() call
    can return RPC_AUTH_GSS, which is passed to request_module().
    
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
    Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>

diff --git a/net/sunrpc/auth_gss/auth_gss.c b/net/sunrpc/auth_gss/auth_gss.c
index 5257d2982ba5..282dfb14db05 100644
--- a/net/sunrpc/auth_gss/auth_gss.c
+++ b/net/sunrpc/auth_gss/auth_gss.c
@@ -1733,6 +1733,7 @@ static void __exit exit_rpcsec_gss(void)
 	rcu_barrier(); /* Wait for completion of call_rcu()'s */
 }
 
+MODULE_ALIAS("rpc-auth-6");
 MODULE_LICENSE("GPL");
 module_param_named(expired_cred_retry_delay,
 		   gss_expired_cred_retry_delay,